go/vt/topo/shard.go

```diff
@@ -284,6 +292,10 @@ func (ts *Server) UpdateShardFields(ctx context.Context, keyspace, shard string,
 // This will lock the Keyspace, as we may be looking at other shard servedTypes.
 // Using GetOrCreateShard is probably a better idea for most use cases.
 func (ts *Server) CreateShard(ctx context.Context, keyspace, shard string) (err error) {
+	if err := ValidateKeyspaceName(keyspace); err != nil {
+		return err
+	}
+
 	// Lock the keyspace, because we'll be looking at ServedTypes.
 	ctx, unlock, lockErr := ts.LockKeyspace(ctx, keyspace, "CreateShard")
```

Comment:
kinda nitpicky, but maybe we should validate the shard name first as well, so we can skip locking entirely if we're just going to fail the call?

Reply:
Ah yeah, let's do that. Fixed!